R/config.R

```diff
@@ -0,0 +1,976 @@
+#'
+#' @description This function returns the specific path to the .git of the
+#' project repository that is being analyzed specified in the input parameter
+#' `config_file`. The input, `config_file` must be a parsed configuration file.
+#' The function will inform the user if the .git path of the project repository
+#' exists in the parsed configuration file, `config_file`.
+#'
+#' @param config_file The parsed configuration file.
+#' @return The local git repository path specified in `config_file`.
+#' @export
+get_git_repo_path <- function(config_file) {
+
+  git_repo_path <- config_file[["version_control"]][["log"]]
+
+  if (is.null(git_repo_path)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(git_repo_path)
+}
+
+#' Returns the list of git branches used for analysis in the current project.
+#'
+#' @description This function returns a list of the git branches used for
+#' analysis in the current project specified in the input parameter
+#' `config_file`. The input, `config_file` must be a parsed configuration file.
+#' The function will inform the user if the list of branches to be analyzed
+#' exists in the parsed configuration file, `config_file`.
+#'
+#' @param config_file The parsed configuration file.
+#' @return The list of git branches.
+#' @export
+get_git_branches <- function(config_file) {
+
+  git_branch <- config_file[["version_control"]][["branch"]]
+
+  if (is.null(git_branch)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(git_branch)
+}
+
+##### Filtering Getter Functions #####
+
+#' Returns the list of file extensions used for filtering files to keep.
+#'
+#' @description This function returns the list of file extensions that will be
+#' used for filtering files specified in the input parameter `config_file`. The
+#' input, `config_file` must be a parsed configuration file. The function will
+#' inform the user if the list of file extensions exists in the parsed
+#' configuration file, `config_file`.
+#'
+#' @param config_file The parsed configuration file.
+#' @return The list of file extensions to keep.
+#' @export
+get_file_extensions <- function(config_file) {
+
+  file_extensions <- config_file[["filter"]][["keep_filepaths_ending_with"]]
+
+  if (is.null(file_extensions)) {
+    warning("Attribute does not exist in the configuration file.")
+  }
+
+  return(file_extensions)
+}
+
+#' Returns the list of file extensions used for filtering files to remove.
+#'
+#' @description This function returns the list of file extensions that will be
+#' used for filtering files specified in the input parameter `config_file`. The
+#' input, `config_file` must be a parsed configuration file. The function will
+#' inform the user if the list of file extensions exists in the parsed
+#' configuration file, `config_file`.
+#'
+#' @param config_file The parsed configuration file.
```

Comment:
Parameters that require input parsed by other functions should always use `\code{\link{}}` please see examples in Kaiaulu codebase. Without that, a user trying to read this get function, will stare at `The parsed configuration file` and have no idea where to look for to create the parameter.